test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1098,6 +1116,39 @@ UCS_TEST_P(test_ucp_am_nbx_rndv, header_and_data_rndv, "RNDV_THRESH=128")
     test_am_send_recv(127, 1);
 }
 
+UCS_TEST_P(test_ucp_am_nbx_rndv, wrong_rndv_cb, "RNDV_THRESH=1")
```

Comment:
let's skip it on valgrind(because it fails with it)
can use
```
UCS_TEST_SKIP_COND_P(test_ucp_am_nbx_rndv, wrong_rndv_cb, RUNNING_ON_VALGRIND)
```
Also please rename the test to `invalid_recv_desc`